src/MusicTheory/NoteOrTuplet.elm

```diff
@@ -0,0 +1,438 @@
+
+
+type NoteOrTuplet
+    = Single DurationAndPitches
+    | Duplet DurationAndPitches DurationAndPitches
+    | Triplet DurationAndPitches DurationAndPitches DurationAndPitches
+    | Quadruplet DurationAndPitches DurationAndPitches DurationAndPitches DurationAndPitches
+    | Quintuplet DurationAndPitches DurationAndPitches DurationAndPitches DurationAndPitches DurationAndPitches
+
+
+type alias DurationAndPitches =
+    { duration : Duration
+    , pitches : List Pitch
+    }
+
+
+type Division
+    = Whole
+    | Half
+    | Quarter
+    | Eighth
+    | Sixteenth
+    | ThirtySecond
+    | SixtyFourth
+    | OneHundredTwentyEighth
+
+
+type Duration
+    = Normal Division TiedOrSeparate
+    | Dotted Division TiedOrSeparate
+    | DoubleDotted Division TiedOrSeparate
+    | TripleDotted Division TiedOrSeparate
+
+
+type TiedOrSeparate
+    = Tied
+    | Separate
+
+
+note : Division -> List Pitch -> NoteOrTuplet
+note division pitchList =
+    Single
+        { duration = Normal division Separate
+        , pitches = pitchList
+        }
+
+
+duplet : Division -> List Pitch -> List Pitch -> NoteOrTuplet
+duplet division pitchList1 pitchList2 =
+    Duplet
+        { duration = Normal division Separate
+        , pitches = pitchList1
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList2
+        }
+
+
+triplet : Division -> List Pitch -> List Pitch -> List Pitch -> NoteOrTuplet
+triplet division pitchList1 pitchList2 pitchList3 =
+    Triplet
+        { duration = Normal division Separate
+        , pitches = pitchList1
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList2
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList3
+        }
+
+
+quadruplet : Division -> List Pitch -> List Pitch -> List Pitch -> List Pitch -> NoteOrTuplet
+quadruplet division pitchList1 pitchList2 pitchList3 pitchList4 =
+    Quadruplet
+        { duration = Normal division Separate
+        , pitches = pitchList1
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList2
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList3
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList4
+        }
+
+
+quintuplet : Division -> List Pitch -> List Pitch -> List Pitch -> List Pitch -> List Pitch -> NoteOrTuplet
+quintuplet division pitchList1 pitchList2 pitchList3 pitchList4 pitchList5 =
+    Quintuplet
+        { duration = Normal division Separate
+        , pitches = pitchList1
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList2
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList3
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList4
+        }
+        { duration = Normal division Separate
+        , pitches = pitchList5
+        }
+
+
+whole : Division
```

Comment:
I think here I would prefer if these constructors return `Duration`. Something like this:

```half : Duration
half =
    Normal Half Separate


quarter : Duration
quarter =
    Normal Quarter Separate```

```tied : Duration -> Duration
withTie duration =
    case duration of
        ...


dotted : Duration -> Duration
dotted duration =
    case duration of
        ...
```

And then use it like this:

```quarter |> tied |> dotted |> Expect.equal (Dotted Quarter Tied)```

Reply:
I think that's fine. But in that case the tuplet constructors would have to change. They currently take a `Division`:
```
triplet : Division -> List Pitch -> List Pitch -> List Pitch -> NoteOrTuplet
```

Reply:
ah right